New resource: `azurerm_logic_app_standard`

dylanmorley opened the pull request:
As per #12023, creating a new resource azurerm_logic_app_standard to support Logic App Standard / Single Tenant model. While this is essentially a _type of_ function_app (Logic app standard is built as an extension on top of the functions infrastructure), there is no attempt at reusing existing code as per the ```function_app_resource```, instead favouring duplication of that code into the new ```azurerm_logic_app_standard```. This is because of rework happening in the web package for 3.0, so duplication was favourable to coupling & there is some behaviour specific to logic apps hosting that has been implemented here. 

* New resource Id validations / parsers etc
* New resource ```azurerm_logic_app_standard``` supporting both 'classic' and 'container' deploy models
* Tests for scenarios + documentation

```
$ TF_ACC=1 go test -v ./internal/services/logic -run=TestAccLogicAppStandard_ -timeout 180m -ldflags="-X=github.com/terraform-providers/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccLogicAppStandard_basic
=== PAUSE TestAccLogicAppStandard_basic
=== RUN   TestAccLogicAppStandard_containerized
=== PAUSE TestAccLogicAppStandard_containerized
=== RUN   TestAccLogicAppStandard_extensionBundle
=== PAUSE TestAccLogicAppStandard_extensionBundle
=== RUN   TestAccLogicAppStandard_requiresImport
=== PAUSE TestAccLogicAppStandard_requiresImport
=== RUN   TestAccLogicAppStandard_tags
=== PAUSE TestAccLogicAppStandard_tags
=== RUN   TestAccLogicAppStandard_tagsUpdate
=== PAUSE TestAccLogicAppStandard_tagsUpdate
=== RUN   TestAccLogicAppStandard_appSettings
=== PAUSE TestAccLogicAppStandard_appSettings
=== RUN   TestAccLogicAppStandard_customShare
=== PAUSE TestAccLogicAppStandard_customShare
=== RUN   TestAccLogicAppStandard_siteConfig
=== PAUSE TestAccLogicAppStandard_siteConfig
=== RUN   TestAccLogicAppStandard_healthCheck
=== PAUSE TestAccLogicAppStandard_healthCheck
=== RUN   TestAccLogicAppStandard_connectionStrings
=== PAUSE TestAccLogicAppStandard_connectionStrings
=== RUN   TestAccLogicAppStandard_updateVersion
=== PAUSE TestAccLogicAppStandard_updateVersion
=== RUN   TestAccLogicAppStandard_3264bit
=== PAUSE TestAccLogicAppStandard_3264bit
=== RUN   TestAccLogicAppStandard_httpsOnly
=== PAUSE TestAccLogicAppStandard_httpsOnly
=== RUN   TestAccLogicAppStandard_createIdentity
=== PAUSE TestAccLogicAppStandard_createIdentity
=== RUN   TestAccLogicAppStandard_updateIdentity
=== PAUSE TestAccLogicAppStandard_updateIdentity
=== RUN   TestAccLogicAppStandard_userAssignedIdentity
=== PAUSE TestAccLogicAppStandard_userAssignedIdentity
=== RUN   TestAccLogicAppStandard_corsSettings
=== PAUSE TestAccLogicAppStandard_corsSettings
=== RUN   TestAccLogicAppStandard_enableHttp2
=== PAUSE TestAccLogicAppStandard_enableHttp2
=== RUN   TestAccLogicAppStandard_minTlsVersion
=== PAUSE TestAccLogicAppStandard_minTlsVersion
=== RUN   TestAccLogicAppStandard_ftpsState
=== PAUSE TestAccLogicAppStandard_ftpsState
=== RUN   TestAccLogicAppStandard_preWarmedInstanceCount
=== PAUSE TestAccLogicAppStandard_preWarmedInstanceCount
=== RUN   TestAccLogicAppStandard_computedPreWarmedInstanceCount
=== PAUSE TestAccLogicAppStandard_computedPreWarmedInstanceCount
=== RUN   TestAccLogicAppStandard_oneIpRestriction
=== PAUSE TestAccLogicAppStandard_oneIpRestriction
=== RUN   TestAccLogicAppStandard_oneServiceTagIpRestriction
=== PAUSE TestAccLogicAppStandard_oneServiceTagIpRestriction
=== RUN   TestAccLogicAppStandard_changeIpToServiceTagIpRestriction
=== PAUSE TestAccLogicAppStandard_changeIpToServiceTagIpRestriction
=== RUN   TestAccLogicAppStandard_oneVNetSubnetIpRestriction
=== PAUSE TestAccLogicAppStandard_oneVNetSubnetIpRestriction
=== RUN   TestAccLogicAppStandard_ipRestrictionRemoved
=== PAUSE TestAccLogicAppStandard_ipRestrictionRemoved
=== RUN   TestAccLogicAppStandard_manyIpRestrictions
=== PAUSE TestAccLogicAppStandard_manyIpRestrictions
=== RUN   TestAccLogicAppStandard_updateStorageAccountKey
=== PAUSE TestAccLogicAppStandard_updateStorageAccountKey
=== RUN   TestAccLogicAppStandard_clientCertMode
=== PAUSE TestAccLogicAppStandard_clientCertMode
=== RUN   TestAccLogicAppStandard_elasticInstanceMinimum
=== PAUSE TestAccLogicAppStandard_elasticInstanceMinimum
=== RUN   TestAccLogicAppStandard_appScaleLimit
=== PAUSE TestAccLogicAppStandard_appScaleLimit
=== RUN   TestAccLogicAppStandard_runtimeScaleMonitoringEnabled
=== PAUSE TestAccLogicAppStandard_runtimeScaleMonitoringEnabled
=== RUN   TestAccLogicAppStandard_dotnetVersion4
=== PAUSE TestAccLogicAppStandard_dotnetVersion4
=== RUN   TestAccLogicAppStandard_dotnetVersion5
=== PAUSE TestAccLogicAppStandard_dotnetVersion5
=== RUN   TestAccLogicAppStandard_dotnetVersion6
=== PAUSE TestAccLogicAppStandard_dotnetVersion6
=== CONT  TestAccLogicAppStandard_basic
=== CONT  TestAccLogicAppStandard_minTlsVersion
=== CONT  TestAccLogicAppStandard_dotnetVersion6
=== CONT  TestAccLogicAppStandard_dotnetVersion5
=== CONT  TestAccLogicAppStandard_dotnetVersion4
--- PASS: TestAccLogicAppStandard_dotnetVersion6 (193.77s)
--- PASS: TestAccLogicAppStandard_basic (216.29s)
=== CONT  TestAccLogicAppStandard_runtimeScaleMonitoringEnabled
--- PASS: TestAccLogicAppStandard_minTlsVersion (253.60s)
=== CONT  TestAccLogicAppStandard_appScaleLimit
--- PASS: TestAccLogicAppStandard_dotnetVersion5 (258.42s)
=== CONT  TestAccLogicAppStandard_elasticInstanceMinimum
--- PASS: TestAccLogicAppStandard_dotnetVersion4 (150.76s)
=== CONT  TestAccLogicAppStandard_clientCertMode
--- PASS: TestAccLogicAppStandard_runtimeScaleMonitoringEnabled (179.30s)
=== CONT  TestAccLogicAppStandard_updateStorageAccountKey
--- PASS: TestAccLogicAppStandard_elasticInstanceMinimum (193.09s)
=== CONT  TestAccLogicAppStandard_manyIpRestrictions
--- PASS: TestAccLogicAppStandard_appScaleLimit (237.52s)
=== CONT  TestAccLogicAppStandard_ipRestrictionRemoved
--- PASS: TestAccLogicAppStandard_manyIpRestrictions (149.70s)
=== CONT  TestAccLogicAppStandard_oneVNetSubnetIpRestriction
--- PASS: TestAccLogicAppStandard_updateStorageAccountKey (339.82s)
=== CONT  TestAccLogicAppStandard_changeIpToServiceTagIpRestriction
--- PASS: TestAccLogicAppStandard_ipRestrictionRemoved (244.77s)
=== CONT  TestAccLogicAppStandard_oneServiceTagIpRestriction
--- PASS: TestAccLogicAppStandard_oneVNetSubnetIpRestriction (168.57s)
=== CONT  TestAccLogicAppStandard_oneIpRestriction
--- PASS: TestAccLogicAppStandard_oneServiceTagIpRestriction (150.69s)
=== CONT  TestAccLogicAppStandard_computedPreWarmedInstanceCount
--- PASS: TestAccLogicAppStandard_oneIpRestriction (150.80s)
=== CONT  TestAccLogicAppStandard_preWarmedInstanceCount
--- PASS: TestAccLogicAppStandard_clientCertMode (592.29s)
=== CONT  TestAccLogicAppStandard_ftpsState
--- PASS: TestAccLogicAppStandard_changeIpToServiceTagIpRestriction (250.52s)
=== CONT  TestAccLogicAppStandard_connectionStrings
--- PASS: TestAccLogicAppStandard_ftpsState (148.47s)
=== CONT  TestAccLogicAppStandard_enableHttp2
--- PASS: TestAccLogicAppStandard_preWarmedInstanceCount (184.16s)
=== CONT  TestAccLogicAppStandard_corsSettings
--- PASS: TestAccLogicAppStandard_computedPreWarmedInstanceCount (229.25s)
=== CONT  TestAccLogicAppStandard_userAssignedIdentity
--- PASS: TestAccLogicAppStandard_connectionStrings (149.97s)
=== CONT  TestAccLogicAppStandard_updateIdentity
--- PASS: TestAccLogicAppStandard_enableHttp2 (149.60s)
=== CONT  TestAccLogicAppStandard_createIdentity
--- PASS: TestAccLogicAppStandard_corsSettings (148.45s)
=== CONT  TestAccLogicAppStandard_httpsOnly
--- PASS: TestAccLogicAppStandard_userAssignedIdentity (209.15s)
=== CONT  TestAccLogicAppStandard_3264bit
--- PASS: TestAccLogicAppStandard_updateIdentity (194.51s)
=== CONT  TestAccLogicAppStandard_updateVersion
--- PASS: TestAccLogicAppStandard_createIdentity (142.31s)
=== CONT  TestAccLogicAppStandard_tagsUpdate
--- PASS: TestAccLogicAppStandard_httpsOnly (135.15s)
=== CONT  TestAccLogicAppStandard_healthCheck
--- PASS: TestAccLogicAppStandard_3264bit (194.51s)
=== CONT  TestAccLogicAppStandard_siteConfig
--- PASS: TestAccLogicAppStandard_updateVersion (197.36s)
=== CONT  TestAccLogicAppStandard_customShare
--- PASS: TestAccLogicAppStandard_healthCheck (152.80s)
=== CONT  TestAccLogicAppStandard_appSettings
--- PASS: TestAccLogicAppStandard_tagsUpdate (194.99s)
=== CONT  TestAccLogicAppStandard_requiresImport
--- PASS: TestAccLogicAppStandard_siteConfig (149.54s)
=== CONT  TestAccLogicAppStandard_tags
--- PASS: TestAccLogicAppStandard_customShare (141.49s)
=== CONT  TestAccLogicAppStandard_extensionBundle
--- PASS: TestAccLogicAppStandard_requiresImport (155.06s)
=== CONT  TestAccLogicAppStandard_containerized
--- PASS: TestAccLogicAppStandard_appSettings (208.52s)
--- PASS: TestAccLogicAppStandard_extensionBundle (142.16s)
--- PASS: TestAccLogicAppStandard_tags (149.48s)
--- PASS: TestAccLogicAppStandard_containerized (145.46s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/logic 1892.801s

```

adamyager commented:
@jackofallops  @dylanmorley  I cannot tell  from what I can see here but does this support App Service Enviroment v3?
The logic apps Product Team tells me its GA for App Service enviroment V3. So if this is included thats great. If not, I can put in an issue for this for a future release

adamyager commented:
ok. I get it. As long as I use app_service_plan_id for the Logic app, it "should" work?

Thanks for the quick response.
We will test

jcanizalez commented:
hey @dylanmorley 

This PR looks great, only a suggestion.
Should be better if you change  kind as a setteable property in the resource.   At the moment you hardcoded the value  **kind := "functionapp,workflowapp"** or functionapp,linux,container,workflowapp based on site config
But for example if you want to deploy a logic app standard on Azure ARC you need to use  **kind := "kubernetes,functionapp,workflowapp,linux"** https://docs.microsoft.com/en-us/azure/logic-apps/azure-arc-enabled-logic-apps-create-deploy-workflows?tabs=azure-cli

So should be useful to allow to set kind property

dylanmorley commented:
@jcanizalez yeah I did think about doing that. However, I went for the resource 'managing' the kind as a design choice - I thought in general it would be preferable to slightly abstract users of the resource from some of the complexities of configuration combinations in order to create the right logic app types

There are couple of examples of this in the resource - I did the same for a number of config settings. Instead of expecting you to have to set app_settings for APP_KIND, workflow bundles etc, the resource creates and manages these for you, which keeps all of these repetitive settings out of everyone's implementations. I thought this would result in slightly cleaner HCL - you'd just need to supply the minimum amount of parameters to get the right type created. Any thoughts on this approach - or would you just prefer to be able to (and _have_ to) specify everything yourself? 

For this scenario, following the same pattern in this PR we could expose ['Extended Location'](https://docs.microsoft.com/en-us/azure/logic-apps/azure-arc-enabled-logic-apps-create-deploy-workflows?tabs=azure-cli#arm-template-1) as a property, and if it's supplied with linuxFxVersion then kind would be "kubernetes,functionapp,workflowapp,linux".

jcanizalez commented:
Thanks for your response @dylanmorley 

Make sense, we can implement support for Logic Apps on Azure ARC later once this PR was release

adamyager commented:
> Thanks for your response @dylanmorley
> 
> Make sense, we can implement support for Logic Apps on Azure ARC later once this PR was release
Is there anything left to get this approved for the current milestone? It was pushed last week and I didnt think anything was left. Love to get my hands on this!!!

djpirra commented:
Is this already available to be used on the latest release? 
If not, when are we supposed to be able to create Logic Apps Standard?

Thank you guys.

dylanmorley commented:
- In latest commits : 
* Added vnet_route_all setting, as per https://github.com/hashicorp/terraform-provider-azurerm/pull/13310
* Added storage_account_share_name to manage WEBSITE_CONTENTSHARE setting, fixes issue with updates
* Updated docs

katbyte commented on internal/services/logic/logic_app_standard_resource.go:
```diff
@@ -0,0 +1,1465 @@
+				Computed: true,
+			},
+
+			"client_cert_mode": {
```
could we un abbreviate this?
```suggestion
			"client_certificate_mode": {
```

dylanmorley replied:
Sure can, will get that done

katbyte commented on internal/services/logic/logic_app_standard_resource.go:
```diff
@@ -0,0 +1,1465 @@
+				Default:  true,
+			},
+
+			"bundle_version_range": {
```
this isn't really a range? so would this make more sense as
```suggestion
			"bundle_version_regex": {
```
or even just
```suggestion
			"bundle_version": {
```

dylanmorley replied:
I think *bundle_version* works - will get that done

katbyte reviewed (changes requested):
Thanks for the PR @dylanmorley - at first glance i just had a couple comment son schema that i've left inline

dylanmorley commented:
Changes as per @katbyte  suggestions done, latest test results

```
$ TF_ACC=1 go test -v ./internal/services/logic -run=TestAccLogicAppStandard_ -timeout 180m -ldflags="-X=github.com/terraform-providers/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccLogicAppStandard_basic
=== PAUSE TestAccLogicAppStandard_basic
=== RUN   TestAccLogicAppStandard_containerized
=== PAUSE TestAccLogicAppStandard_containerized
=== RUN   TestAccLogicAppStandard_extensionBundle
=== PAUSE TestAccLogicAppStandard_extensionBundle
=== RUN   TestAccLogicAppStandard_siteConfigVnetRouteAllEnabled
=== PAUSE TestAccLogicAppStandard_siteConfigVnetRouteAllEnabled
=== RUN   TestAccLogicAppStandard_appSettingsVnetRouteAllEnabled
=== PAUSE TestAccLogicAppStandard_appSettingsVnetRouteAllEnabled
=== RUN   TestAccLogicAppStandard_requiresImport
=== PAUSE TestAccLogicAppStandard_requiresImport
=== RUN   TestAccLogicAppStandard_tags
=== PAUSE TestAccLogicAppStandard_tags
=== RUN   TestAccLogicAppStandard_tagsUpdate
=== PAUSE TestAccLogicAppStandard_tagsUpdate
=== RUN   TestAccLogicAppStandard_appSettings
=== PAUSE TestAccLogicAppStandard_appSettings
=== RUN   TestAccLogicAppStandard_customShare
=== PAUSE TestAccLogicAppStandard_customShare
=== RUN   TestAccLogicAppStandard_siteConfig
=== PAUSE TestAccLogicAppStandard_siteConfig
=== RUN   TestAccLogicAppStandard_healthCheck
=== PAUSE TestAccLogicAppStandard_healthCheck
=== RUN   TestAccLogicAppStandard_connectionStrings
=== PAUSE TestAccLogicAppStandard_connectionStrings
=== RUN   TestAccLogicAppStandard_updateVersion
=== PAUSE TestAccLogicAppStandard_updateVersion
=== RUN   TestAccLogicAppStandard_3264bit
=== PAUSE TestAccLogicAppStandard_3264bit
=== RUN   TestAccLogicAppStandard_httpsOnly
=== PAUSE TestAccLogicAppStandard_httpsOnly
=== RUN   TestAccLogicAppStandard_createIdentity
=== PAUSE TestAccLogicAppStandard_createIdentity
=== RUN   TestAccLogicAppStandard_updateIdentity
=== PAUSE TestAccLogicAppStandard_updateIdentity
=== RUN   TestAccLogicAppStandard_corsSettings
=== PAUSE TestAccLogicAppStandard_corsSettings
=== RUN   TestAccLogicAppStandard_enableHttp2
=== PAUSE TestAccLogicAppStandard_enableHttp2
=== RUN   TestAccLogicAppStandard_minTlsVersion
=== PAUSE TestAccLogicAppStandard_minTlsVersion
=== RUN   TestAccLogicAppStandard_ftpsState
=== PAUSE TestAccLogicAppStandard_ftpsState
=== RUN   TestAccLogicAppStandard_preWarmedInstanceCount
=== PAUSE TestAccLogicAppStandard_preWarmedInstanceCount
=== RUN   TestAccLogicAppStandard_computedPreWarmedInstanceCount
=== PAUSE TestAccLogicAppStandard_computedPreWarmedInstanceCount
=== RUN   TestAccLogicAppStandard_oneIpRestriction
=== PAUSE TestAccLogicAppStandard_oneIpRestriction
=== RUN   TestAccLogicAppStandard_oneServiceTagIpRestriction
=== PAUSE TestAccLogicAppStandard_oneServiceTagIpRestriction
=== RUN   TestAccLogicAppStandard_changeIpToServiceTagIpRestriction
=== PAUSE TestAccLogicAppStandard_changeIpToServiceTagIpRestriction
=== RUN   TestAccLogicAppStandard_oneVNetSubnetIpRestriction
=== PAUSE TestAccLogicAppStandard_oneVNetSubnetIpRestriction
=== RUN   TestAccLogicAppStandard_ipRestrictionRemoved
=== PAUSE TestAccLogicAppStandard_ipRestrictionRemoved
=== RUN   TestAccLogicAppStandard_manyIpRestrictions
=== PAUSE TestAccLogicAppStandard_manyIpRestrictions
=== RUN   TestAccLogicAppStandard_updateStorageAccountKey
=== PAUSE TestAccLogicAppStandard_updateStorageAccountKey
=== RUN   TestAccLogicAppStandard_clientCertMode
=== PAUSE TestAccLogicAppStandard_clientCertMode
=== RUN   TestAccLogicAppStandard_elasticInstanceMinimum
=== PAUSE TestAccLogicAppStandard_elasticInstanceMinimum
=== RUN   TestAccLogicAppStandard_appScaleLimit
=== PAUSE TestAccLogicAppStandard_appScaleLimit
=== RUN   TestAccLogicAppStandard_runtimeScaleMonitoringEnabled
=== PAUSE TestAccLogicAppStandard_runtimeScaleMonitoringEnabled
=== RUN   TestAccLogicAppStandard_dotnetVersion4
=== PAUSE TestAccLogicAppStandard_dotnetVersion4
=== RUN   TestAccLogicAppStandard_dotnetVersion5
=== PAUSE TestAccLogicAppStandard_dotnetVersion5
=== RUN   TestAccLogicAppStandard_dotnetVersion6
=== PAUSE TestAccLogicAppStandard_dotnetVersion6
=== CONT  TestAccLogicAppStandard_basic
=== CONT  TestAccLogicAppStandard_enableHttp2
=== CONT  TestAccLogicAppStandard_siteConfig
=== CONT  TestAccLogicAppStandard_corsSettings
--- PASS: TestAccLogicAppStandard_corsSettings (206.24s)
=== CONT  TestAccLogicAppStandard_updateIdentity
--- PASS: TestAccLogicAppStandard_siteConfig (206.91s)
=== CONT  TestAccLogicAppStandard_createIdentity
--- PASS: TestAccLogicAppStandard_enableHttp2 (207.21s)
=== CONT  TestAccLogicAppStandard_httpsOnly
--- PASS: TestAccLogicAppStandard_basic (209.46s)
=== CONT  TestAccLogicAppStandard_3264bit
--- PASS: TestAccLogicAppStandard_createIdentity (151.10s)
=== CONT  TestAccLogicAppStandard_updateVersion
--- PASS: TestAccLogicAppStandard_httpsOnly (155.46s)
=== CONT  TestAccLogicAppStandard_connectionStrings
--- PASS: TestAccLogicAppStandard_3264bit (216.85s)
=== CONT  TestAccLogicAppStandard_healthCheck
--- PASS: TestAccLogicAppStandard_updateIdentity (277.38s)
=== CONT  TestAccLogicAppStandard_manyIpRestrictions
--- PASS: TestAccLogicAppStandard_connectionStrings (155.29s)
=== CONT  TestAccLogicAppStandard_dotnetVersion6
--- PASS: TestAccLogicAppStandard_updateVersion (203.23s)
=== CONT  TestAccLogicAppStandard_dotnetVersion5
--- PASS: TestAccLogicAppStandard_healthCheck (157.22s)
=== CONT  TestAccLogicAppStandard_dotnetVersion4
--- PASS: TestAccLogicAppStandard_manyIpRestrictions (158.74s)
=== CONT  TestAccLogicAppStandard_runtimeScaleMonitoringEnabled
--- PASS: TestAccLogicAppStandard_dotnetVersion6 (161.34s)
=== CONT  TestAccLogicAppStandard_appScaleLimit
--- PASS: TestAccLogicAppStandard_dotnetVersion5 (148.98s)
=== CONT  TestAccLogicAppStandard_elasticInstanceMinimum
--- PASS: TestAccLogicAppStandard_dotnetVersion4 (148.49s)
=== CONT  TestAccLogicAppStandard_clientCertMode
--- PASS: TestAccLogicAppStandard_appScaleLimit (187.37s)
=== CONT  TestAccLogicAppStandard_updateStorageAccountKey
--- PASS: TestAccLogicAppStandard_runtimeScaleMonitoringEnabled (242.19s)
=== CONT  TestAccLogicAppStandard_requiresImport
--- PASS: TestAccLogicAppStandard_elasticInstanceMinimum (180.08s)
=== CONT  TestAccLogicAppStandard_customShare
--- PASS: TestAccLogicAppStandard_customShare (155.93s)
=== CONT  TestAccLogicAppStandard_appSettings
--- PASS: TestAccLogicAppStandard_requiresImport (167.67s)
=== CONT  TestAccLogicAppStandard_tagsUpdate
--- PASS: TestAccLogicAppStandard_clientCertMode (339.39s)
=== CONT  TestAccLogicAppStandard_tags
--- PASS: TestAccLogicAppStandard_updateStorageAccountKey (292.89s)
=== CONT  TestAccLogicAppStandard_oneIpRestriction
--- PASS: TestAccLogicAppStandard_tags (155.00s)
=== CONT  TestAccLogicAppStandard_ipRestrictionRemoved
--- PASS: TestAccLogicAppStandard_tagsUpdate (201.23s)
=== CONT  TestAccLogicAppStandard_oneVNetSubnetIpRestriction
--- PASS: TestAccLogicAppStandard_appSettings (215.92s)
=== CONT  TestAccLogicAppStandard_changeIpToServiceTagIpRestriction
--- PASS: TestAccLogicAppStandard_oneIpRestriction (154.81s)
=== CONT  TestAccLogicAppStandard_oneServiceTagIpRestriction
--- PASS: TestAccLogicAppStandard_oneVNetSubnetIpRestriction (177.40s)
=== CONT  TestAccLogicAppStandard_siteConfigVnetRouteAllEnabled
--- PASS: TestAccLogicAppStandard_ipRestrictionRemoved (256.36s)
=== CONT  TestAccLogicAppStandard_appSettingsVnetRouteAllEnabled
--- PASS: TestAccLogicAppStandard_changeIpToServiceTagIpRestriction (264.34s)
=== CONT  TestAccLogicAppStandard_preWarmedInstanceCount
--- PASS: TestAccLogicAppStandard_siteConfigVnetRouteAllEnabled (156.62s)
=== CONT  TestAccLogicAppStandard_computedPreWarmedInstanceCount
--- PASS: TestAccLogicAppStandard_appSettingsVnetRouteAllEnabled (146.81s)
=== CONT  TestAccLogicAppStandard_extensionBundle
--- PASS: TestAccLogicAppStandard_oneServiceTagIpRestriction (366.94s)
=== CONT  TestAccLogicAppStandard_containerized
--- PASS: TestAccLogicAppStandard_preWarmedInstanceCount (182.83s)
=== CONT  TestAccLogicAppStandard_ftpsState
--- PASS: TestAccLogicAppStandard_computedPreWarmedInstanceCount (178.00s)
=== CONT  TestAccLogicAppStandard_minTlsVersion
--- PASS: TestAccLogicAppStandard_extensionBundle (146.46s)
--- PASS: TestAccLogicAppStandard_containerized (159.44s)
--- PASS: TestAccLogicAppStandard_ftpsState (158.32s)
--- PASS: TestAccLogicAppStandard_minTlsVersion (160.13s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/logic 1947.060s

```

mbfrahry reviewed (approved):
LGTM!

dylanmorley commented:
Thanks @mbfrahry. I'm working extensively  in this area at the moment, so any bugs or issues reported in this resource, tag me & I'll have a look.

raptor75 commented:
When will azurerm_app_service_virtual_network_swift_connection be updated to support azurerm_logic_app_standard @dylanmorley please?

djpirra commented:
Oh no. It’s not supported in this release?
If that is the case, there is any alternative other than manual assignment?

On Fri, 1 Oct 2021 at 9:37 PM raptor75 ***@***.***> wrote:

> When will azurerm_app_service_virtual_network_swift_connection be updated
> to support azurerm_logic_app_standard @dylanmorley
> <https://github.com/dylanmorley> please?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/hashicorp/terraform-provider-azurerm/pull/13196#issuecomment-932428262>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADZ5Q5UWUJUS3WHYGN4VXOLUEXWXFANCNFSM5DEXZOHQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>
-- 
Melhores Cumprimentos,
Luis Simões

raptor75 commented:
@dylanmorley @djpirra 

Never mind, just noticed that the documentation for azurerm_logic_app_standard mentions it, but the documentation for azurerm_app_service_virtual_network_swift_connection is not yet updated.

I am testing it myself right now anyway, hoping it works.

raptor75 commented:
@djpirra it worked!

djpirra commented:
Thanks for the good news!
Awesome work guys.

On Fri, 1 Oct 2021 at 10:25 PM raptor75 ***@***.***> wrote:

> @djpirra <https://github.com/djpirra> it worked!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/hashicorp/terraform-provider-azurerm/pull/13196#issuecomment-932457980>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADZ5Q5R5NDSWN5FWUJP5AVTUEX4JHANCNFSM5DEXZOHQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>
-- 
Melhores Cumprimentos,
Luis Simões

raptor75 commented:
@dylanmorley I am facing an issue where the storage account blob structure is not consistently being created correctly when using this resource to create the logic app standard. This does not happen with function app. Is this a known issue please?

dylanmorley commented:
@raptor75 Not anything I'm aware of, but please do [open an issue](https://github.com/hashicorp/terraform-provider-azurerm/issues/new?assignees=&labels=&template=Bug_Report.md) and mention me in it, I'll have a look.
